Fix #7858: hide rocketinsights when disabled

Miraeld opened the pull request:
# Description

Fixes #7858 
This pull request introduces a feature flag to control the visibility and activation of the Rocket Insights addon and related notices throughout the admin interface. The changes ensure that Rocket Insights features, notices, and dashboard elements are only shown when the addon is enabled, improving flexibility for different site configurations and user roles. Tests and templates have been updated to support and validate the new feature flag.

## Type of change

- [x] Enhancement (non-breaking change which improves an existing functionality).

## Detailed scenario

### What was tested
- First install while Rocket Insights is disabled : No notice is shown related to Rocket Insights
<img width="768" height="237" alt="Screenshot 2025-11-27 at 05 11 35" src="https://github.com/user-attachments/assets/c1ed2799-4f99-4afd-88d7-facdf8b59e21" />

- Loading WP Rocket dashboard while RocketInsights is disabled: No mention of Rocket insights
<img width="1468" height="914" alt="Screenshot 2025-11-27 at 05 11 49" src="https://github.com/user-attachments/assets/e344c548-2dbb-4143-9365-e3ddc8815044" />


### How to test

Adding the snippet `add_filter( 'rocket_rocket_insights_enabled', '__return_false' );` to disable Rocket Insights. Then install/load any page of WP Rocket.

## Technical description

### Documentation

**Feature flag implementation and conditional display:**

* Replaced previous reseller/non-live checks with a new `is_allowed()` feature flag in the Rocket Insights context for controlling the display of addon status and promotion notices (`Controller.php`, `Subscriber.php`). [[1]](diffhunk://#diff-643fec80a4c58eef07379deef4b2645dc54d96651d55e6cd21bc2a1bbc094d75L50-R51) [[2]](diffhunk://#diff-c69013204dfdf12389e54172d012e1c15b1fcdb2fd0771cb4f05c8fa6fb35857R533-R537)
* Added the `rocket_insights_enabled` value, determined by the feature flag, to dashboard section data for conditional rendering in the dashboard template (`Page.php`, `dashboard.php`). [[1]](diffhunk://#diff-1f8a05a3d25f121c99e4c387a399983759c6fe94f8b72d480dfe1e7978789241R441) [[2]](diffhunk://#diff-f8d59140ce75fbb164412f589d5135f0ab87fe85448ca3b0f0433c2cbecc694aR46) [[3]](diffhunk://#diff-f8d59140ce75fbb164412f589d5135f0ab87fe85448ca3b0f0433c2cbecc694aR57)
* Updated the thank-you license notice to use a new filter (`rocket_rocket_insights_enabled`) to determine whether to show Rocket Insights messaging, with fallback messaging if disabled (`notices.php`). [[1]](diffhunk://#diff-3d873eb27942f2237fec3111334ee8726a01efd731f36b90e2f4616daf2907c1R456-R466) [[2]](diffhunk://#diff-3d873eb27942f2237fec3111334ee8726a01efd731f36b90e2f4616daf2907c1R476-R485)

**Testing and validation:**

* Added new fixture and integration tests to verify correct behavior when Rocket Insights is disabled, including filter injection for `rocket_rocket_insights_enabled` (`DisplayAddonStatusTest.php`, `MaybeDisplayRocketInsightsPromotionNotice.php`). [[1]](diffhunk://#diff-097d034af47c41a8d4668ebdc90f24078502c22ef5037660fb4a72563bed8f4fR33-R44) [[2]](diffhunk://#diff-9d9f26f6e3798cdb4523af3d35f600a473b4d31e66cab5128cb7dc5d7d2758c3R54-R64) [[3]](diffhunk://#diff-295f198e29fe51a8b5d9f19def1fdd3eebeff5113abcb52952fad8d455885220R28-R33) [[4]](diffhunk://#diff-5d0210a1e438ed82c4604e19d9b79b0bc7f40734447b11073789723776639fdfR76-R81)

These changes ensure that Rocket Insights features are only surfaced to users when explicitly enabled, and provide hooks for further customization or integration.

# Mandatory Checklist

## Code validation

- [x] I validated all the Acceptance Criteria. If possible, provide screenshots or videos.
- [x] I triggered all changed lines of code at least once without new errors/warnings/notices.
- [x] I implemented built-in tests to cover the new/changed code.

## Code style

- [x] I wrote a self-explanatory code about what it does.
- [x] I protected entry points against unexpected inputs.
- [x] I did not introduce unnecessary complexity.
- [x] Output messages (errors, notices, logs) are explicit enough for users to understand the issue and are actionnable.

Miraeld committed:
Fix #7858: Hide Rocket Insights UI when disabled via filter

- Add is_allowed() check to Controller::display_addon_status()
- Add is_allowed() check to Subscriber::maybe_display_rocket_insights_promotion_notice()
- Conditionally show Rocket Insights mention in rocket_thank_you_license()
- Add rocket_insights_enabled flag to dashboard section data
- Conditionally display Rocket Insights mention in activation notice

Miraeld committed:
Add tests for Rocket Insights disabled filter

- Add testShouldNotDisplayWhenRocketInsightsDisabled to DisplayAddonStatusTest
- Add testShouldNotDisplayNoticeWhenRocketInsightsDisabled to MaybeDisplayRocketInsightsPromotionNotice
- Add filter handling in integration tests

Miraeld committed:
Fix phpcs

jeawhanlee commented on inc/admin/ui/notices.php:
```diff
@@ -453,15 +453,36 @@ function rocket_thank_you_license() {
+			WP_ROCKET_PLUGIN_NAME,
+			'</strong>'
+			);
+		}
```
```suggestion
        $message = sprintf(
		/* translators: %1$s = <strong>, %2$s = plugin name, %3$s = </strong> */
		__( '%1$s %2$s is good to go! %3$s Your website is already faster.', 'rocket' ),
		'<strong>',
		WP_ROCKET_PLUGIN_NAME,
		'</strong>'
		);
		
		// This filter is documented in inc/Engine/Admin/RocketInsights/Context/Context.php.
		$rocket_insights_enabled = wpm_apply_filters_typed( 'boolean', 'rocket_rocket_insights_enabled', true );

		if ( $rocket_insights_enabled ) {
			$message = sprintf(
			/* translators: %1$s = plugin name, %2$s = opening link tag, %3$s = closing link tag */
			__( '%1$s %2$s is good to go! %3$s Your website is already faster. Visit %4$s Rocket Insights %5$s to check your homepage\'s performance, add more pages to measure WP Rocket\'s impact, and keep your site fast.', 'rocket' ),
			'<strong>',
			WP_ROCKET_PLUGIN_NAME,
			'</strong>',
			'<a href="' . admin_url( 'options-general.php?page=' . WP_ROCKET_PLUGIN_SLUG . '&rocket_source=notice_thankyou_license#rocket_insights' ) . '">',
			'</a>'
			);
		}
```

Miraeld committed:
Apply feedback

Miraeld committed:
Merge branch 'develop' into fix/7858-hide-rocketinsights-when-disabled

Mai-Saad commented:
Related TP https://wpmediaqa.testrail.io/index.php?/runs/view/1162&page_called_by=run_view&group_by=cases:section_id&group_order=asc&display=tree

hanna-meda commented:
Hi @Miraeld, thank you for this PR.

Everything checks out except the RI notices. Those are still showing for resellers and on local environments. We already had an open issue for that (#7941). If this will be fixed here, we should close that ticket as duplicate. cc @piotrbak 
Sending the ticket back to In Progress based on this comment.

Miraeld committed:
Fix: Hide Rocket Insights notices for resellers and non-live sites

Apply rocket_rocket_insights_enabled filter in Subscriber to disable RI for reseller accounts and non-live installations. This ensures the filter remains the single source of truth for RI visibility across all features (notices, settings, etc).

Fixes #7941

Miraeld commented:
Hello @hanna-meda , 
Thank you for you feedback !

I've pushed a commit
**What was done:**

Added a new filter callback `maybe_disable_for_reseller_or_non_live()` that hooks into `rocket_rocket_insights_enabled`
This callback checks `is_reseller_or_non_live()` and returns false when it's a reseller account OR non-live site
The filter is applied early, making it the single source of truth for RI visibility
**Why this approach:**
- **Centralized** logic - The reseller/non-live check lives in the RocketInsights module where it belongs
- **Filter as single source of truth** - All code respecting `rocket_rocket_insights_enabled` now automatically respects reseller/non-live restrictions
- **Clean separation** - No need to inject container or RI Context into global notice functions
- **Consistent behavior** - The thank-you notice, settings section, promotion notice, and dashboard widget all respect the same rules
- **Respects user overrides** - If someone explicitly sets the filter to false, it stays false

Miraeld committed:
Merge branch 'develop' into fix/7858-hide-rocketinsights-when-disabled

jeawhanlee reviewed (approved):
I like the thinking here, Sir

hanna-meda reviewed (approved):
QA done. Everything is working as expected now. 

[testrail-report-797.pdf](https://github.com/user-attachments/files/24080634/testrail-report-797.pdf)
